src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -177,5 +209,338 @@ private void ThrowIfDisposed()
+                }
+            }
+
+            if (metrics[0].EventQueueTime > TimeSpan.Zero && metrics[0].EventQueueTime < metrics[NumberOfSamplesToConsider - 1].EventQueueTime)
+            {
+                bool queueTimeIncreasing =
+                    IsTrueForLastN(
+                        metrics,
+                        NumberOfSamplesToConsider,
+                        (prev, next) => prev.EventQueueTime <= next.EventQueueTime);
+                if (queueTimeIncreasing)
+                {
+                    status.Vote = ScaleVote.ScaleOut;
+                    _logger.LogInformation($"Queue time is increasing for '{_entityPath}'.");
+                    return status;
+                }
+            }
+
+            bool queueLengthDecreasing =
+                IsTrueForLastN(
+                    metrics,
+                    NumberOfSamplesToConsider,
+                    (prev, next) => prev.TotalNewEventCount > next.TotalNewEventCount);
+            if (queueLengthDecreasing)
+            {
+                status.Vote = ScaleVote.ScaleIn;
+                _logger.LogInformation($"Queue length is decreasing for '{_entityPath}'.");
+                return status;
+            }
+
+            bool queueTimeDecreasing = IsTrueForLastN(
+                metrics,
+                NumberOfSamplesToConsider,
+                (prev, next) => prev.EventQueueTime > next.EventQueueTime);
+            if (queueTimeDecreasing)
+            {
+                status.Vote = ScaleVote.ScaleIn;
+                _logger.LogInformation($"Queue time is decreasing for '{_entityPath}'.");
+                return status;
+            }
+
+            _logger.LogInformation($"Service Bus entity '{_entityPath}' is steady.");
+
+            return status;
+        }
+
+        private bool TimeToLogWarning()
+        {
+            DateTime currentTime = DateTime.UtcNow;
+            bool timeToLog = currentTime >= _nextWarningTime;
+            if (timeToLog)
+            {
+                _nextWarningTime = currentTime.AddHours(1);
+            }
+            return timeToLog;
+        }
+
+        private void ParseTopicAndSubscription(string entityPath, out string topic, out string subscription)
+        {
+            string divider = "/subscriptions/";
+
+            int anchor = entityPath.IndexOf(divider, StringComparison.InvariantCultureIgnoreCase);
+            if (anchor < 0)
+            {
+                throw new InvalidOperationException($"{entityPath} is either formatted incorectly, or is not a valid Service Bus subscription path");
+            }
+
+            topic = entityPath.Substring(0, anchor);
+            subscription = entityPath.Substring(anchor + divider.Length, entityPath.Length - (anchor + divider.Length));
+        }
+
+        private static bool IsTrueForLastN(IList<ServiceBusTriggerMetrics> samples, int count, Func<ServiceBusTriggerMetrics, ServiceBusTriggerMetrics, bool> predicate)

```

Comment:
I realize the various implementations are repeating this helper :) I though about adding it to the SDK as a helper, but didn't want to expand the surface area unecessarily. We might add helpers like this in the future if they are common.